### PR TITLE
OS400: Update strings for ccsid-ifier

### PR DESCRIPTION
**Pull request by NattyNarwhal**
Fixes build.

Sidebar: IBM doesn't support or offer QADRT anymore. In the future, it may be better to use something like the compiler's own `LOCALETYPE(*LOCALEUTF)` option, which switches string constants to UTF-8 and has its own UTF-8 handling version of some (but not all) libc functions.

### CODE REVIEW

File: packages/OS400/chkstrings.c [hunk 1]
```diff
@@ -34,7 +34,7 @@
  * made, the EXPECTED_STRING_LASTZEROTERMINATED/EXPECTED_STRING_LAST
  * values can be updated to match the latest enum values in urldata.h.
  */
-#define EXPECTED_STRING_LASTZEROTERMINATED  (STRING_TEMP_URL + 1)
+#define EXPECTED_STRING_LASTZEROTERMINATED  (STRING_DNS_LOCAL_IP6 + 1)
```
Comment (bagder) [inline packages/OS400/chkstrings.c]: Why not use `STRING_LASTZEROTERMINATED` for this?

Comment (jay) [reply packages/OS400/chkstrings.c]: It's purposely like that to check for runtime inconsistencies I think.

/cc @jonrumsey

Comment (jonrumsey) [reply packages/OS400/chkstrings.c]: Yes, the #define in chkstrings.c is expected to be be manually updated whenever STRING_LASTZEROTERMINATED is updated. The value is compared as part of the OS/400 build process to detect a potential missing code change to the EBCDIC<->ASCII string conversion in curl_easy_setopt_ccsid().

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**